base/src/types.rs

```diff
@@ -533,6 +566,17 @@ impl<Id, T> Type<Id, T>
     }
 }
 
+pub trait TypeRef: Sized {
```

Comment:
It would need to be implemented on `ArcType` so that the sharing can be preserved. I suppose implementing it on `ArcType` is fine now but if we add more type representations a trait is the way to go to prevent duplication. Should I change it now?

#143